packaging.md

```diff
@@ -0,0 +1,156 @@
+# set the entry point (application name)
+CMD ["<executable-name>"]
+```
+
+To create a local Docker image from the `Dockerfile` use the `docker build` command from the application's source location, e.g.:
+
+```bash
+$ docker build . -t <my-app>:<my-app-version>
+```
+
+To test the local image use the `docker run` command, .e.g.:
+
+```bash
+$ docker run <my-app>:<my-app-version>
+```
+
+Finally, use the `docker push` command to publish the application's Docker image to a Docker repository of your choice. e.g.:
+
+```bash
+$ docker tag <my-app>:<my-app-version> <docker-hub-user>/<my-app>:<my-app-version>
+$ docker push <docker-hub-user>/<my-app>:<my-app-version>
+```
+
+At this point, the application's Docker image is ready to be deployed to the server hosts (which need to run docker), or to one of the platforms that supports Docker deployments.
+
+See [Docker's documentation](https://docs.docker.com/engine/reference/commandline/) for more complete information about Docker.
+
+### Distroless
+
+[Distroless](https://github.com/GoogleContainerTools/distroless) is a project by Google that attempts to create minimal images containing only the application and its runtime dependencies. They do not contain package managers, shells or any other programs you would expect to find in a standard Linux distribution.
+
+Since distroless supports Docker and is based on Debian, packaging a Swift application on it is fairly similar to the Docker process above. Here is an example `Dockerfile` that builds and packages the application on top of a distroless's C++ base image:
+
+```Dockerfile
+#------- build -------
+# Building using Ubuntu Bionic since its compatible with Debian runtime
+FROM swiftlang/swift:nightly-bionic as builder
+
+# set up the workspace
+RUN mkdir /workspace
+WORKDIR /workspace
+
+# copy the source to the docker image
+COPY . /workspace
+
+RUN swift build -c release -Xswiftc -g
+
+#------- package -------
+# Running on distroless C++ since it includes
+# all(*) the runtime dependencies Swift programs need
+FROM gcr.io/distroless/cc-debian10
+# copy executables
+COPY --from=builder /workspace/.build/release/<executable-name> /
+# copy Swift's dynamic libraries dependencies
+COPY --from=builder /usr/lib/swift/linux/lib*so* /
+
+# set the entry point (application name)
+CMD ["<executable-name>"]
+```
+
+Note the above uses `gcr.io/distroless/cc-debian10` as the runtime image which should work for Swift programs that do not use `FoundationNetworking` or `FoundationXML`. In order to provide more complete support we (the community) could put in a PR into distroless to introduce a base image for Swift that includes `libcurl` and `libxml` which are required for `FoundationNetworking` and `FoundationXML` respectively.
+
+## Archive (Tarball, ZIP file, etc.)
+
+Since cross-compiling Swift for Linux is not [yet] supported on Mac or Windows, we need to use virtualization technologies like Docker to compile applications we are targeting to run on Linux.
```

Comment:
```suggestion
Since cross-compiling Swift for Linux is not (yet) supported on Mac or Windows, we need to use virtualization technologies like Docker to compile applications we are targeting to run on Linux.
```

?